src/Psl/Type/Internal/DictType.php

```diff
@@ -44,15 +44,19 @@ public function coerce(mixed $value): array
 
         $trace = $this->getTrace();
         $key_type = $this->key_type->withTrace($trace->withFrame('dict<' . $this->key_type->toString() . ', _>'));
-        $value_type = $this->value_type->withTrace($trace->withFrame('dict<_, ' . $this->value_type->toString() . '>'));
-
         $result = [];
 
         /**
          * @var Tk $k
          * @var Tv $v
          */
         foreach ($value as $k => $v) {
+            $value_type = $this->value_type->withTrace(
+                $trace->withFrameAtPath(
+                    'dict<_, ' . $this->value_type->toString() . '>',
```

Comment:
string concatenations are fast, but maybe it helps to build the frame name outside of the loop and use it instead of always recalculating the frame name?
It will always be the same value.